lib/opensearch/transport/client.rb

```diff
@@ -183,13 +186,22 @@ def initialize(arguments = {}, &block)
       end
 
       # Performs a request through delegation to {#transport}.
-      def perform_request(method, path, params = {}, body = nil, headers = nil)
+      def perform_request(method, path, params = {}, body = nil, headers = {})
         method = @send_get_body_as if method == 'GET' && body
         if (opaque_id = params.delete(:opaque_id))
```

Comment:
Not relevant to your PR, but I think this is broken 😅 (sym vs string key)

Reply:
Nice catch! I'll fix that!

Reply:
Tho I think this feature should be removed for 4.x for the same reason that we moved `ignore 404` feature to the client level. Users wanting to use this feature can already do so by assigning that value to the `X-Opaque-Id` header directly when calling the API.